cli/tests/unit/flock_test.ts

```diff
@@ -17,125 +17,115 @@ unitTest(
 );
 
 async function runFlockTests(opts: { sync: boolean }) {
-  assertEquals(
-    await checkFirstBlocksSecond({
-      firstExclusive: true,
-      secondExclusive: false,
-      sync: opts.sync,
-    }),
-    true,
-    "exclusive blocks shared",
-  );
-  assertEquals(
-    await checkFirstBlocksSecond({
-      firstExclusive: false,
-      secondExclusive: true,
+  const filePath = "cli/tests/testdata/fixture.json";
+
+  await checkExclusiveLock();
+  await checkNonExclusiveLock();
+
+  async function checkExclusiveLock() {
+    const fileLock = createFileLock({
+      filePath,
+      exclusive: true,
       sync: opts.sync,
-    }),
-    true,
-    "shared blocks exclusive",
-  );
-  assertEquals(
-    await checkFirstBlocksSecond({
-      firstExclusive: true,
-      secondExclusive: true,
+    });
+    try {
+      await fileLock.waitEnterLock();
+      assertEquals(
+        checkFileCanRead(filePath),
+        false,
```

Comment:
Looks like I didn't actually fix this because it doesn't work this way on Linux.